lib/oj/bag.rb

```diff
@@ -37,9 +36,9 @@ def respond_to?(m)
     # @raise [NoMethodError] if the instance variable is not defined.
     def method_missing(m, *args, &block)
       raise ArgumentError.new("wrong number of arguments (#{args.size} for 0) to method #{m}") unless args.nil? or args.empty?
-      at_m = ('@' + m.to_s).to_sym
-      raise NoMethodError.new("undefined method #{m}", m) unless instance_variable_defined?(at_m)
-      instance_variable_get(at_m)
+      result = instance_variable_get(:"@#{m}")
+      raise NoMethodError.new("undefined method #{m}", m) unless result
+      result
     end
```

Comment:
Problem here. The reason the instance_variable_defined? is called is that is $VERBOSE is true you get errors displayed on the console. I often run in that mode to catch errors in code when I am developing. I presume others do to to catch mistakes. The old was is better in this case. I'm not sure if '@' + m.to_s is more efficient than "@#{m}". Probably doesn't make much of a difference. Either is okay with me.